src/compiler/semantic/rules.rs

```diff
@@ -336,11 +336,87 @@ fn types_rule_tl(
     }
 }
 
+/// Check if the condition in an if statement is a logic expression (allowing bool literals and
+/// signals).
+fn if_condition_rule(analyser: &mut Analyser, stmt: &Statement<BigInt, Identifier>) {
+    match stmt {
+        Statement::IfThen(dsym, cond, _) | Statement::IfThenElse(dsym, cond, _, _) => {
+            check_cond_for_bool(analyser, cond, dsym);
+        }
+        _ => {}
+    }
+}
+
+/// Helper function to check if an expression is a logic expression.
+fn check_cond_for_bool(
+    analyser: &mut Analyser,
+    expr: &Expression<BigInt, Identifier>,
+    dsym: &DebugSymRef,
+) {
+    use Expression::*;
+
+    match expr {
+        // Check boolean literals
+        True(_) | False(_) => {}
+
+        // Check if it is a boolean signal
+        Query(_, id) => {
+            if let Some(symbol) = analyser.symbols.find_symbol(&analyser.cur_scope, id.name()) {
+                if symbol.symbol.get_type() != "bool" {
+                    analyser.error(
+                        format!(
+                            "Signal in condition for the if statement must be bool, found {:#?}",
+                            expr
+                        ),
+                        dsym,
+                    )
+                }
+            } else {
+                analyser.error(
+                    format!(
+                        "Condition in if statement uses undeclared variable {:#?}",
```

Comment:
this error is capture already in another checked, we should just not do anything if it is not find, as it is outside this scope of responsability